index-advisor.md

```diff
@@ -0,0 +1,222 @@
++----+---------------------+---------------------+-------------+------------+------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------+-------+
+```
+
+### `RECOMMEND INDEX` command options
+
+You can configure and view options for the `RECOMMEND INDEX` command to fine-tune its behavior for your workloads using the following statements:
+
+```sql
+RECOMMEND INDEX SET <option> = <value>;
+RECOMMEND INDEX SHOW OPTION;
+```
+
+The following options are available:
+
+- `timeout`: specifies the maximum time allowed for running the `RECOMMEND INDEX` command.
+- `max_num_index`: defines the maximum number of indexes to include in the result of `RECOMMEND INDEX`.
+- `max_index_columns`: sets the maximum number of columns allowed in multi-column indexes in the result.
+- `max_num_query`: specifies the maximum number of queries to select from the statement summary workload.
+
+To check your current option settings, use the `RECOMMEND INDEX SHOW` command:
+
+```sql
+mysql> RECOMMEND INDEX SHOW OPTION;
++-------------------+-------+---------------------------------------------------------+
+| option            | value | description                                             |
++-------------------+-------+---------------------------------------------------------+
+| max_num_index     | 5     | The maximum number of indexes to recommend.             |
+| max_index_columns | 3     | The maximum number of columns in an index.              |
+| max_num_query     | 1000  | The maximum number of queries to recommend indexes.     |
+| timeout           | 30s   | The timeout of index advisor.                           |
++-------------------+-------+---------------------------------------------------------+
+4 rows in set (0.00 sec)
+```
+
+To modify an option, use the `RECOMMEND INDEX SET` command. For example, to change the `timeout` option:
+
+```sql
+mysql> RECOMMEND INDEX SET timeout='20s';
+Query OK, 1 row affected (0.00 sec)
+```
+
+### Limitations
+
+The index recommendation feature has the following limitations, which will be addressed in future releases:
+
+- It does not support prepared statements. The `RECOMMEND INDEX RUN` command cannot recommend indexes for queries executed through the `Prepare` and `Execute` protocol.
+- It does not provide recommendations for deleting indexes. Future updates will include logic for [removing indexes](#remove-unused-indexes) in the `RECOMMEND` command.
+- A user interface (UI) for the Index Advisor is not yet available but will be introduced in the future.
```

Comment:
```suggestion
- Currently, it does not support [prepared statements](/develop/dev-guide-prepared-statement.md)[Prepared Statements](/develop/dev-guide-prepared-statement.md). The `RECOMMEND INDEX RUN` statement cannot recommend indexes for queries executed through the `Prepare` and `Execute` protocol.
- Currently, it does not provide recommendations for deleting indexes.
- Currently, a user interface (UI) for the Index Advisor is not yet available.
```

Reply:
If no specific customer requirements, can we remove them to avoid any future promises? 

https://developers.google.cn/style/future

Avoid trying to document future features or products, even in innocuous ways. Don't pre-announce anything in documentation unless it has been approved by your legal counsel.